### DISCUSSION TITLE
Using eslint and flat config without package.json

### DISCUSSION THREAD

**Discussion by marlun** [eslint/eslint#17021] [Q&A]
I'm working on a JavaScript project without a build process so I don't need or want a package.json. If I add a eslint.config.js to my root project folder and add the following:
```javascript
import js from "@eslint/js";

export default [js.configs.recommended];
```
I get an error: **Warning: To load an ES module, set "type": "module" in the package.js on or use the .mjs extension.** I don't think I'm allowed to use eslint.config.mjs right because if I change it eslint doesn't seem to find it. Can I solve it or is package.json a requirement for using eslint?

**Comment by Rec0iL99** [comment]
Hi @marlun, yes `package.json` is a requirement for ESLint to work as expected. You can find the [requirement](https://eslint.org/docs/latest/use/getting-started#manual-set-up) in the docs. ESLint is only [officially supported](https://eslint.org/docs/latest/use/getting-started#prerequisites) for the Nodejs runtime and it's always recommended to use `package.json` when using Nodejs.

Thanks for using ESLint.

**Comment by marlun** [comment]
Thanks for answering! Isn't it a reasonable suggestion that you should be able to use it without a package.json? I mean, I understand that Node.js is required since eslint uses it but there has to be thousands of tools in npm which you can install globally and then use everywhere without the need of a package.json in the folder where you use it? Is there more that is required from the package.json except for the `"type": "module"` property for it to work?

**Reply by Rec0iL99** [reply]
You can use ESLint by [installing it globally](https://eslint.org/docs/latest/use/getting-started#global-install), but any plugins/shareable configs that you want to use in your project should be installed **locally** for it to work as expected. Not using a package.json in your project would mean you need to manage your dependencies manually.

**Reply by nzakas** [reply]
This is not about requiring `package.json`, but rather, about how Node.js determines whether a JavaScript file is an ESM formatted file or a CommonJS formatted file. ESLint itself doesn't care, but Node.js assumes any `.js` file is CommonJS by default, so you need to tell Node.js that it's not either by using `package.json`, renaming to `.mjs`, or I think there may still be a command line flag as well.
